app/controllers/region_controller.rb

```diff
@@ -0,0 +1,60 @@
+class RegionController < ApplicationController
+  before_filter :load_vars
+
+  def show
+    region_hash = {
+      eez_area: 0,
+      ts_area: 0,
+      pa_land_area: 0,
+      pa_marine_area: 0,
+      percentage_pa_land_cover: 0,
+      percentage_pa_eez_cover: 0,
+      percentage_pa_ts_cover: 0,
+      land_area: 0,
+      percentage_pa_cover: 0,
+      pa_eez_area: 0,
+      pa_ts_area: 0,
+      percentage_pa_marine_cover: 0,
+      marine_area: 0,
+      polygons_count: 0,
+      points_count: 0
+    }
+
+    Country.joins(:region).where(region_id: @region.id).each do |country|
+      region_hash[:eez_area] += country.statistic.eez_area || 0
+      region_hash[:ts_area] += country.statistic.ts_area || 0
+      region_hash[:pa_land_area] += country.statistic.pa_land_area || 0
+      region_hash[:pa_marine_area] += country.statistic.pa_marine_area || 0
+      region_hash[:percentage_pa_land_cover] += country.statistic.percentage_pa_land_cover || 0
+      region_hash[:percentage_pa_eez_cover] += country.statistic.percentage_pa_eez_cover || 0
+      region_hash[:percentage_pa_ts_cover] += country.statistic.percentage_pa_ts_cover || 0
+      region_hash[:land_area] += country.statistic.land_area || 0
+      region_hash[:percentage_pa_cover] += country.statistic.percentage_pa_cover || 0
+      region_hash[:pa_eez_area] += country.statistic.pa_eez_area || 0
+      region_hash[:pa_ts_area] += country.statistic.pa_ts_area || 0
+      region_hash[:percentage_pa_marine_cover] += country.statistic.percentage_pa_marine_cover || 0
+      region_hash[:marine_area] += country.statistic.marine_area || 0
+      region_hash[:polygons_count] += country.statistic.polygons_count || 0
+      region_hash[:points_count] += country.statistic.points_count || 0
+    end
```

Comment:
Instead of creating an hash with all the possible values and then going through each attribute to initialise them like this in the controller, you could create a model that calculates just the needed  stats.
For example, you can have a look at the `CountryStatistic` model.
In the model then, you can add functions to calculate only the needed stats; we don't necessarily need to replicate the entire structure of the country (for now at least)

Reply:
@Levia originally I had planned to create a model called `RegionStatistic` which would store all of the necessary statistics inside a database table. I had originally created a migration for this, but went against it after some discussion, because I thought maybe I could use Redis for this. However, should I create a migration and a table for these statistics called `region_statistics`?

Reply:
Sorry, I mistaken `CoutryPresenter` with `CountryStatistic`.
With countries we are given a file with the stats and we import those every month.
With regions, we just need to do some calculations over those country stats; that said, I think it should be enough to have a `RegionPresenter` that uses the `CountryStatistic` model for each country the region contains.
So no need for migrations as `RegionPresenter` in not an ActiveRecord model.
Does this make sense to you?

Reply:
Thanks @Levia I totally understand what you mean!